src/skmultiflow/trees/hoeffding_adaptive_tree_regressor.py

```diff
@@ -165,12 +165,12 @@ def _partial_fit(self, X, y, weight):
         self.sum_of_values += weight * y
         self.sum_of_squares += weight * y * y
 
-        try:
-            self.sum_of_attribute_values = np.add(self.sum_of_attribute_values, np.multiply(weight, X))
-            self.sum_of_attribute_squares = np.add(self.sum_of_attribute_squares, np.multiply(weight, np.power(X, 2)))
-        except ValueError:
-            self.sum_of_attribute_values = np.multiply(weight, X)
-            self.sum_of_attribute_squares = np.multiply(weight, np.power(X, 2))
+        if self.sum_of_attribute_values.size != 0:
```

Comment:
Using `try` is a trick to improve performance when there are few chances of the error being raised. In this case, the `if` condition will only be evaluated as `False` once for initialization. Thus the usage of `try`.

Reply:
Ah, ok. We used the same technique in embedded systems.
Should we add a comment to make it clear that we know when and why the `try` fails?

Reply:
Hi @AlbandeCrevoisier, that's a great idea.